lasy/utils/grid.py

```diff
@@ -67,9 +67,9 @@ def __init__(self, dim, lo, hi, npoints, n_azimuthal_modes=None, is_envelope=Tru
             ncomp = 2 * self.n_azimuthal_modes - 1
             self.shape = (ncomp, self.npoints[0], self.npoints[1])
 
-        self.temporal_field = np.zeros(self.shape, dtype="complex128")
         # This overwrites the dtype of temporal_field above
         self.set_is_envelope(is_envelope)
+        self.temporal_field = np.zeros(self.shape, dtype=self.dtype)
```

Comment:
This reordering conveniently initialize the array after the type has been selected.
We can now delete the comment in line 70.